File: 1. Two Sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        res = []
        lookup = dict()
        
        for i in range(len(nums)):
            complement = target - nums[i]
            
            if (complement in lookup):
                return [lookup[complement], i]
            else:
                lookup[nums[i]] = i
        return None

File: 1. Two Sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();

        unordered_map<int, int> sum_map;

        for (int i = 0; i < n; i++) {
            if (sum_map.find(target - nums[i]) == sum_map.end())
                sum_map[nums[i]] = i;
            else
                return {sum_map[target - nums[i]], i};
        }

        return {-1, -1};
    }
};
